code/modules/mob/living/simple_animal/hostile/mining_mobs/gutlunch.dm

```diff
@@ -64,54 +68,71 @@
 		return TRUE
 
 	if(isobj(the_target) && is_type_in_typecache(the_target, wanted_objects))
-		return TRUE
+		if(udder.reagents.total_volume < udder.reagents.maximum_volume)
+			return TRUE
+		else
+			return FALSE
 
 	return FALSE
 
-/mob/living/simple_animal/hostile/asteroid/gutlunch/regenerate_icons(new_udder_volume, max_udder_volume)
+/mob/living/simple_animal/hostile/asteroid/gutlunch/regenerate_icons()
 	cut_overlays()
 	var/static/gutlunch_full_overlay
 	if(isnull(gutlunch_full_overlay))
 		gutlunch_full_overlay = iconstate2appearance(icon, "gl_full")
-	if(new_udder_volume == max_udder_volume)
+	if(udder.reagents.total_volume == udder.reagents.maximum_volume && health > 0)
 		add_overlay(gutlunch_full_overlay)
 	..()
 
+/mob/living/simple_animal/hostile/asteroid/gutlunch/death()
+	. = ..()
+	chem_splash(get_turf(src), 1, list(udder.reagents))
```

Comment:
instead, make a subtype of udder component that listens for death (via the signal) and spills the reagents. Or just add it to all udders